Cargo.toml

```diff
@@ -41,10 +41,12 @@ spv-out = ["spirv"]
 wgsl-in = ["codespan-reporting"]
 wgsl-out = []
 hlsl-out = []
+error-reporting = ["codespan-reporting"]
```

Comment:
I made the reporting optional but I'm not sure it is the best way forward.
Firstly, it requires `required-features` to be added to `bin/naga.rs` so every invocation must use `cargo run --features error-reporting,...`, which is a bit annoying. Also, I agree with @JCapucho that it isn't the responsibility of the library but rather of the user to handle error reporting (maybe someone already has their own infrastructure for that or want's more flexibilty).

Maybe moving the error reporting to `bin/naga.rs` and having no feature to enable it in `naga` itself would be the better move. What do you think?

Reply:
If we can move it fully into bin/naga, that's absolutely the best, yes.

Reply:
I moved it to naga.rs but there is one annoying issue, there seems to be no way to have dependencies (codespan-reporting) for `bin`s: https://stackoverflow.com/questions/35711044/how-can-i-specify-binary-only-dependencies.
The solutions listed on stack overflow are 
1. move the executable to an example
2. use required-features and a feature just for the bin
3. make the binary it's own package

Reply:
I see. We've been fighting with it quite a bit.
Sounds like we still need the `glsl-error-reporting` feature (note the glsl in the name) then.

Reply:
For what its worth, for option 3. with a separate `naga-cli` crate, `cargo run` would still work in root directory if we specify 
```toml
[workspace]
members = [".", "naga-cli"]
default-members = ["naga-cli"]
```

https://doc.rust-lang.org/cargo/reference/workspaces.html#package-selection

Reply:
oh interesting, I didn't know a "default-members" is a thing!
The other problem with moving it into a crate is that essentially we'd have to duplicate all the features.
Overall, it's a whole different discussion. Let's have it, but maybe separately from this PR.

Reply:
Opened a PR for discussion: #938